### ISSUE TITLE
nn.softmax does not take dimension

### ISSUE THREAD

**Issue by mohammad-py** [pytorch/pytorch#1688]
One thing which is so valuable in some problems, is that we are able to apply the softmax operation along a predefined dimension. [Tesnorflow](https://www.tensorflow.org/api_docs/python/tf/nn/softmax) has this parameter in definition of the softmax layer but pytorch has not. Good to have this feature in next release of pytorch!

**Comment by apaszke** [comment]
Duplicate of #1020
